### PR TITLE
Add language support for Go workspaces

### PR DESCRIPTION
**Pull request by goto1134**
## Description
Adds support for Go workspace ([`go.work`](https://go.dev/blog/get-familiar-with-workspaces#workspaces)) files.

## Checklist:

- [x] **I am adding a new extension to a language.**
  - [x] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      
      - https://github.com/search?type=code&q=NOT+is%3Afork+path%3Ago.work.sum
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/portainer/portainer/blob/develop/go.work.sum
    - Sample license(s): zlib License
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A%2F%28%5E%7C%5C%2F%29go%5C.work%24%2F
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/portainer/portainer/blob/develop/go.work
    - Sample license(s): zlib License
  - [ ] I have included a syntax highlighting grammar: [URL to grammar repo]
      
  - [x] I have added a color
     - Hex value: `#00ADD8`
     - Rationale: To match other Go language files 
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: grammars.yml [hunk 1]
```diff
@@ -1065,6 +1065,7 @@ vendor/grammars/vscode-gleam:
 vendor/grammars/vscode-go:
 - go.mod
 - go.sum
+- go.work
```
Comment (lildude) [inline grammars.yml:1068]: 🤔 where did this come from? I don't see it in the grammar. This file is kept up to date based on the content of the grammars and shouldn't be manually modified.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -2450,6 +2453,17 @@ Go Module:
...
   language_id: 947461016
+Go Workspace:
+  type: data
+  color: "#00ADD8"
+  aliases:
+  - go.work
+  - go work
+  filenames:
+  - go.work
+  tm_scope: go.work
```
Comment (lildude) [inline lib/linguist/languages.yml:2464]: This won't work if a grammar doesn't provide this scope. As these files are structured like `go.mod` files so using that scope should do the trick.

Comment (goto1134) [reply lib/linguist/languages.yml:2464]: Oh, thank you for clarifying that! Will fix the scope ASAP

File: grammars.yml (hunk 1 above)
Comment (goto1134) [reply grammars.yml]: Removed the non-existing grammar here

**Review by lildude — APPROVED** [review]
LGTM. Thanks.

Note: this PR will not be merged until close to when the next release is made. See [here](https://github.com/github/linguist/blob/master/docs/troubleshooting.md#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes) for more details.

**Pull request merged**